src/daos-serialize/daos-serialize.c

```diff
@@ -0,0 +1,226 @@
+        {0                      , 0                , 0, 0  }
+    };
+
+    /* Parse options */
+    unsigned long long bytes = 0;
+    int usage = 0;
+    while(1) {
+        int c = getopt_long(
+                    argc, argv, "vh",
+                    long_options, &option_index
+                );
+
+        if (c == -1) {
+            break;
+        }
+
+        switch(c) {
+            case 'v':
+                mfu_debug_level = MFU_LOG_VERBOSE;
+                break;
+            case 'h':
+                usage = 1;
+                break;
+            case '?':
+                usage = 1;
+                break;
+            default:
+                if(rank == 0) {
+                    printf("?? getopt returned character code 0%o ??\n", c);
+                }
+        }
+    }
+
+    /* If we need to print the usage
+     * then do so before internal processing */
+    if (usage) {
+        if (rank == 0) {
+            print_usage();
+        }
+        mfu_finalize();
+        MPI_Finalize();
+        return 1;
+    }
+
+    char** argpaths = (&argv[optind]);
+    
+    /* The remaining arguments are treated as src/dst paths */
+    int numpaths = argc - optind;
+
+    /* advance to next set of options */
+    optind += numpaths;
+
+    /* Before processing, make sure we have at least two paths to begin with */
+    if (numpaths < 1) {
+        MFU_LOG(MFU_LOG_ERR, "At least one pool and container is required:"
+                MFU_ERRF, MFU_ERRP(-MFU_ERR_INVAL_ARG));
+        mfu_finalize();
+        MPI_Finalize();
+        return 1;
+    }
+
+    /* TODO: support input file rather than required all pool/cont paths
+     * be input on cmd line */
+    /* TODO daos_init
+     *      gather paths on rank 0 and put in flist
+     *      mfu_flist_summarize
+     *      mfu_flist_spread
+     *      loop over local size list and connect to pool/cont,
+     *      then call cont_serialize_hdlr
+     *      disconnect form pool/cont
+     *      daos_fini
+     */
+
+    /* create an empty file list */
+    mfu_flist tmplist = mfu_flist_new();
+
+    rc = daos_init();
+    if (rc != 0) {
+        MFU_LOG(MFU_LOG_ERR, "Failed to initialize daos");
+        return 1;
+    }
+
+    /* use rank 0's paths, and spread them evenly among ranks
+     * Each "path" is a DAOS pool/cont, and this tool will
+     * serialize each container into its own hdf5 file */
+    int i;
+    if (rank == 0) {
+        for (i = 0; i < numpaths; i++) {
+                uint64_t idx = mfu_flist_file_create(tmplist);
+                mfu_flist_file_set_cont(tmplist, idx, argpaths[i]);
+        }
+    }
+
+    mfu_flist_summarize(tmplist);
+    mfu_flist newflist = mfu_flist_spread(tmplist);
+
+    /* get size of local list for each rank */
+    uint64_t size = mfu_flist_size(newflist);
+
+    uuid_t pool_uuids[size];
+    uuid_t cont_uuids[size];
+    daos_handle_t pohs[size];
+    daos_handle_t cohs[size];
```

Comment:
@dsikich In the future, we can consider adjust the existing `daos_args` to hold multiple pools/conts. It would take some refactoring of the existing functions, but it would also allow us to support multiple inputs for `dcp`, and would generally centralize the code.